src/mesh_smoothing.rs

```diff
@@ -0,0 +1,582 @@
+            (18, 17, 16),
+            (14, 10, 17),
+            (3, 9, 0),
+            (0, 1, 2),
+            (2, 3, 4),
+            (7, 9, 18),
+            (6, 1, 9),
+            (6, 18, 1),
+            (15, 0, 8),
+            (15, 8, 9),
+            (1, 18, 11),
+            (11, 12, 13),
+            (13, 1, 10),
+            (2, 4, 15),
+            (1, 15, 3),
+            (8, 2, 14),
+            (8, 6, 5),
+            (16, 10, 14),
+            (16, 14, 1),
+            (8, 5, 7),
+            (18, 8, 17),
+            (17, 7, 6),
+            (11, 13, 16),
+            (18, 16, 12),
+            (10, 11, 17),
+        ];
+
+        (faces, vertices)
+    }
+
+    fn shape_for_smoothing_with_anchors() -> (Vec<(u32, u32, u32)>, Vertices) {
+        let vertices = vec![
+            Point3::new(30.21796, -6.119943, 0.0),
+            Point3::new(32.031532, 1.328689, 0.0),
+            Point3::new(33.875141, -3.522298, 3.718605),
+            Point3::new(34.571838, -2.071111, 2.77835),
+            Point3::new(34.778172, -5.285372, 3.718605),
+            Point3::new(36.243252, -3.80194, 3.718605),
+            Point3::new(36.741604, -10.146505, 0.0),
+            Point3::new(39.676025, 1.905633, 0.0),
+            Point3::new(42.587009, -5.186427, 0.0),
+        ];
+
+        let faces = vec![
+            (4, 8, 5),
+            (4, 6, 8),
+            (5, 8, 7),
+            (3, 5, 7),
+            (0, 2, 1),
+            (1, 2, 3),
+            (0, 4, 2),
+            (1, 3, 7),
+            (0, 6, 4),
+            (2, 4, 5),
+            (2, 5, 3),
+        ];
+
+        (faces, vertices)
+    }
+
+    fn shape_for_smoothing_with_anchors_50_iterations() -> (Vec<(u32, u32, u32)>, Vertices) {
+        let vertices = vec![
+            Point3::new(30.21796, -6.119943, 0.0),
+            Point3::new(32.031532, 1.328689, 0.0),
+            Point3::new(34.491065, -2.551039, 0.0),
+            Point3::new(36.00632, -0.404003, 0.0),
+            Point3::new(36.372859, -5.260642, 0.0),
+            Point3::new(37.826656, -2.299296, 0.0),
+            Point3::new(36.741604, -10.146505, 0.0),
+            Point3::new(39.676025, 1.905633, 0.0),
+            Point3::new(42.587009, -5.186427, 0.0),
+        ];
+
+        let faces = vec![
+            (4, 8, 5),
+            (4, 6, 8),
+            (5, 8, 7),
+            (3, 5, 7),
+            (0, 2, 1),
+            (1, 2, 3),
+            (0, 4, 2),
+            (1, 3, 7),
+            (0, 6, 4),
+            (2, 4, 5),
+            (2, 5, 3),
+        ];
+
+        (faces, vertices)
+    }
+
+    #[test]
+    fn test_laplacian_smoothing_preserves_face_vertex_normal_count() {
+        let (faces, vertices) = torus();
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_computed_normals(
+            faces.clone(),
+            vertices.clone(),
+            NormalStrategy::Sharp,
+        );
+        let relaxed_geometry_1 = laplacian_smoothing(&geometry, 1);
```

Comment:
Let's also test 0. Apart from not being useful, the alorithm has no reason to break apart for 0 iterations. The usual algorithmic test cases are 0, 1 and a really big number (10 suffices as that in this case).